src/Padrightpropertiesdraft.agda

```diff
@@ -0,0 +1,105 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of padRight for Vec
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Padrightpropertiesdraft where
+
+open import Data.Fin.Base using (Fin; zero; suc; inject≤)
+open import Data.Nat.Base using (ℕ; zero; suc; _+_; _≤_; z≤n; s≤s; _∸_)
+open import Data.Nat.Properties using (≤-refl; ≤-trans; m≤m+n; m+n∸m≡n)
+open import Data.Vec.Base
+open import Data.Vec.Properties using (map-replicate; zipWith-replicate; padRight-trans)
+open import Function.Base using (_∘_; _$_)
+open import Level using (Level)
+open import Relation.Binary.PropositionalEquality.Core
+  using (_≡_; refl; cong; sym; trans; subst)
+open import Relation.Binary.PropositionalEquality.Properties
+  using (module ≡-Reasoning)
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    m n p : ℕ
+
+
+------------------------------------------------------------------------
+-- Interaction with map
+
+padRight-map : ∀ (f : A → B) (m≤n : m ≤ n) (a : A) (xs : Vec A m) →
+               map f (padRight m≤n a xs) ≡ padRight m≤n (f a) (map f xs)
+padRight-map f z≤n a [] = map-replicate f a _
+padRight-map f (s≤s m≤n) a (x ∷ xs) = cong (f x ∷_) (padRight-map f m≤n a xs)
+
+------------------------------------------------------------------------
+-- Interaction with lookup
+
+padRight-lookup : ∀ (m≤n : m ≤ n) (a : A) (xs : Vec A m) (i : Fin m) →
+                  lookup (padRight m≤n a xs) (inject≤ i m≤n) ≡ lookup xs i
```

Comment:
Feels like this should be derived via lemmas instead of directly by induction.

`lookup ys (inject ≤ i m≤n)` is `lookup (take i ys) i` and `take i (padRight m≤n a xs)` is `xs`.